### ISSUE TITLE
Laravel 5.2 Transport Error

### ISSUE THREAD

**Issue by comancsm** [elasticquent/Elasticquent#19]
Hello,

I just installed the package and ES 2.1.1.  The server is running but i get an empty response at localhost:9200.(i use homestead)

When i tried to create the index, i get the error below

 ' ' '
ErrorException in Client.php line 69:
Argument 1 passed to Elasticsearch\Client::__construct() must be an instance of Elasticsearch\Transport, array given, called in /home/vagrant/Code/Watch/vendor/elasticquent/elasticquent/src/ElasticquentTrait.php on line 60 and defined
' ' '

**Comment by comancsm** [comment]
Edit: Server is working ,still get the same error. Is it suposed to work with ES version 2.1.1?

Thanks

**Comment by timgws** [comment]
There is no reason why it shouldn't. Give me 24 hours to set it up in my lab and test it.





---- On Sat, 16 Jan 2016 00:08:29 +1100 notifications@github.com wrote ---- Edit: Server is working ,still get the same error. Is it suposed to work with ES version 2.1.1?
Thanks

**Comment by comancsm** [comment]
Hi, thanks for the quick reply. From what i've read, the 2.0 API is different from the 1.X one.

My IDE tells me at  \Elasticsearch\Client($config); : "Required Parameter $endpoint missing"

// ElastiquentTrait.php
 public function getElasticSearchClient()
    {
        $config = $this->getElasticConfig();

        return new \Elasticsearch\Client($config); <-------- $endpoint missing
    }

**Comment by timgws** [comment]
That sounds very much like you don't have the most up to date version of the package.
Make sure that you have Dev-master in your composer.json file, and run composer update please.




---- On Sat, 16 Jan 2016 01:39:32 +1100 notifications@github.com wrote ---- Hi, thanks for the quick reply. From what i've read, the 2.0 API is different from the 1.X one.
My IDE tells me at  \Elasticsearch\Client($config); : "Required Parameter $endpoint missing"
// ElastiquentTrait.php

 public function getElasticSearchClient()

    {

        $config = $this->getElasticConfig();
    return new \Elasticsearch\Client($config); <-------- $endpoint missing
}

**Comment by comancsm** [comment]
Yes, that was it. Thanks for the quick reply. Highly appreciate it.

Do you by any chance have experience with the homestead setup? I can access localhost:9200 and from my browser and it works but within my app i get an error:

TransportException in Connection.php line 201:
cURL error 7: Failed to connect to localhost port 9200: Connection refused
in Connection.php line 201
at Connection->Elasticsearch\Connections\{closure}(array('transfer_stats' => array('url' => 'http://localhost:9200/default', 'content_type' => null, 'http_code' => '0', 'header_size' => '0', 'request_size' => '0', 'filetime' => '-1', 'ssl_verify_result' => '0', 'redirect_count' => '0', 'total_time' => '0.000163', 'namelookup_time' => '1.9E-5', 'connect_time' => '0', 'pretransfer_time' => '0', 'size_upload' => '0', 'size_download' => '0', 'speed_download' => '0', 'speed_upload' => '0', 'download_content_length' => '-1', 'upload_content_length' => '-1', 'starttransfer_time' => '0', 'redirect_time' => '0', 'redirect_url' => '', 'primary_ip' => '', 'certinfo' => array(), 'primary_port' => '0', 'local_ip' => '', 'local_port' => '0', 'error' => 'Failed to connect to localhost port 9200: Connection refused', 'errno' => '7'), 'curl' => array('error' => 'Failed to connect to localhost port 9200: Connection refused', 'errno' => '7'), 'effective_url' => 'http://localhost:9200/default', 'status' => null, 'reason' => null, 'body' => null, 'headers' => array(), 'error' => object(ConnectException))) in FulfilledPromise.php line 25

**Comment by timgws** [comment]
Is the elasticsearch server running inside or outside of the container? It is 3am where I am now, but I will try to write a quick response.
Homestead runs on a separate IP on a separate network. Try using your network interfaces IP address instead of localhost. localhost inside homestead is not the same as localhost on your physical machine. This is because homestead usually starts a virtual machine.






---- On Sat, 16 Jan 2016 02:27:35 +1100 notifications@github.com wrote ---- Yes, that was it. Thanks for the quick reply. Highly appreciate it.
Do you by any chance have experience with the homestead setup? I can access localhost:9200 and from my browser and it works but within my app i get an error:
TransportException in Connection.php line 201:

cURL error 7: Failed to connect to localhost port 9200: Connection refused

in Connection.php line 201

at Connection->Elasticsearch\Connections{closure}(array('transfer_stats' => array('url' => 'http://localhost:9200/default', 'content_type' => null, 'http_code' => '0', 'header_size' => '0', 'request_size' => '0', 'filetime' => '-1', 'ssl_verify_result' => '0', 'redirect_count' => '0', 'total_time' => '0.000163', 'namelookup_time' => '1.9E-5', 'connect_time' => '0', 'pretransfer_time' => '0', 'size_upload' => '0', 'size_download' => '0', 'speed_download' => '0', 'speed_upload' => '0', 'download_content_length' => '-1', 'upload_content_length' => '-1', 'starttransfer_time' => '0', 'redirect_time' => '0', 'redirect_url' => '', 'primary_ip' => '', 'certinfo' => array(), 'primary_port' => '0', 'local_ip' => '', 'local_port' => '0', 'error' => 'Failed to connect to localhost port 9200: Connection refused', 'errno' => '7'), 'curl' => array('error' => 'Failed to connect to localhost port 9
 200: Con
 nection refused', 'errno' => '7'), 'effective_url' => 'http://localhost:9200/default', 'status' => null, 'reason' => null, 'body' => null, 'headers' => array(), 'error' => object(ConnectException))) in FulfilledPromise.php line 25

**Comment by comancsm** [comment]
Oh, i;m sorry, just reply whenever you can.

Homestead is running in a vagrant box, the same as the laravel application.

Here is my elasticsearch.php config.

'config' => [
        'hosts'     => ['10.0.2.15:9200'],
        'logging'   => false,
        'logPath'   => storage_path() . '/logs/elasticsearch.log',
        'logLevel'  => Monolog\Logger::WARNING,
    ],

When i ssh into the vagrant box, and i curl into http://10.0.2.15:9200 it works as well. I tried all variations in the config, the thrown error is still on: 

at Connection->Elasticsearch\Connections{closure}(array('transfer_stats' => array('url' => 'http://localhost:9200/default', 'con

.

**Comment by timgws** [comment]
Ok. I will try it in my lab a little later today and see what happens.




---- On Sat, 16 Jan 2016 02:45:29 +1100 notifications@github.com wrote ---- Oh, i;m sorry, just reply whenever you can.
Homestead is running in a vagrant box, the same as the laravel application.
Here is my elasticsearch.php config.
'config' => [

        'hosts'     => ['10.0.2.15:9200'],

        'logging'   => false,

        'logPath'   => storage_path() . '/logs/elasticsearch.log',

        'logLevel'  => Monolog\Logger::WARNING,

    ],
When i ssh into the vagrant box, and i curl into http://10.0.2.15:9200 it works as well. I tried all variations in the config, the thrown error is still on: 
at Connection->Elasticsearch\Connections{closure}(array('transfer_stats' => array('url' => 'http://localhost:9200/default', 'con
.

**Comment by petercoles** [comment]
@comancsm I'm able to connect from an app in homestead to a remote elastic database, so it certainly works.

Your error shows that your app isn't picking up your config settings and is trying unsuccessfully to connect to an elastic cluster local to your homestead box. So the issue is why are your configs being ignored.

One possibility could be that you've called the file elasticsearch.php, but Elasticquent by default is looking for a file called elasticquent.php. If it's not that, the following information would be helpful for debugging: the location of your config file, what steps you're taking to make your app aware of the name and path of the file, and what version of Laravel you're using.

**Comment by timgws** [comment]
I am with @petercoles... I can't see any issue here.

**Comment by putnik** [comment]
I had the same error, but in Laravel 5.1. I fixed it by adding `"elasticsearch/elasticsearch": "^1"` into composer.json. So, it maybe not connected with Laravel 5.2, but with Elasticsearch-PHP 2.*.
